src/ConvertCommand.php

```diff
@@ -29,26 +29,100 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $this->loadConfig($input);
 
-        $filename = $input->getArgument('filename');
-        $cards = json_decode(file_get_contents($filename));
+        $filename = $input->getArgument('source');
 
-        if (!is_array($cards)) {
-            throw new \Exception(sprintf('Could not read parsed vCards from %s', $filename));
+        // we want to check for image upload show stoppers as early as possible
+        if ($input->getOption('image')) {
+            $this->uploadImagePreconditions($this->config['fritzbox'], $this->config['phonebook']);
         }
 
-        // filter
-        $filters = $this->config['filters'];
-        $filtered = filter($cards, $filters);
+        error_log("Reading vCard(s) from file " . $filename);
+        $local = localProvider($filename);
+
+        $progress = new ProgressBar($output);
+        $progress->start();
+        $vcards = download($local, [], function () use ($progress) {
+            $progress->advance();
+        });
+        $progress->finish();
+
+        $quantity = count($vcards);
+        error_log(sprintf("\nRead %d vCard(s)", $quantity));
+
+        // image upload
+        if ($input->getOption('image')) {
+            error_log("Detaching and uploading image(s)");
 
-        if ($json = $input->getOption('raw')) {
-            file_put_contents($json, json_encode($filtered, self::JSON_OPTIONS));
```

Comment:
warum entfällt die "raw" Option?

Reply:
Mit der Nutzung von sabre/vobject ergibt sich aus meiner Sicht keine Notwendigkeit mehr weitere Datenformen (JSON) in den Zwischenstufen zu verwenden.
Download: einlesen vom Server und als Ergebnis ein Plain VCF mit serialize()
Convert: einlesen der VCF und im Ergebis XML (Phonebook-Format)
Upload: einlesen XML und upload